httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/bootstrap/H2ServerBootstrap.java

```diff
@@ -163,6 +163,7 @@ public final H2ServerBootstrap setVersionPolicy(final HttpVersionPolicy versionP
      * @return this instance.
      */
     public final H2ServerBootstrap setH2Config(final H2Config h2Config) {
+        Args.check(!h2Config.isPushEnabled(), "A server MUST NOT set enable_push according to RFC-9113 6.5.2.");
```

Comment:
@CoolTomatos Can we drop the reference to a specific RFC and a specific section? RFC do get revised and updated. I do not want to comb for specific RFC numbers in our code base